back end/clustering.py

```diff
@@ -0,0 +1,359 @@
+# This file gets all the sentences from txt files with texts.
+# It then divides the text in sentences and sentences into words
+ 
+from __future__ import absolute_import, division, print_function
+import codecs
+#finds all pathnames matching a pattern, like regex
+import glob
+#log events for libraries
+import logging
+#concurrency
+import multiprocessing
+#dealing with operating system , like reading file
+import os
+#pretty print, human readable
+import pprint
+#regular expressionss
+import re
+#natural language toolkit
+import nltk
+#word 2 vec
+import gensim.models.word2vec as w2v
+from gensim import models
+#dimensionality reduction
+import sklearn.manifold
+#math
+import numpy as np
+#plotting
+import matplotlib.pyplot as plt
+#parse dataset
+import pandas as pd
+#visualization
+import seaborn as sns
+import pickle
+from scipy import spatial
+from sklearn.cluster import KMeans
+
+# Download tokenizers
+# nltk.download("punkt") # pretrained tokenizer
+# nltk.download("stopwords") # remove words like and, the, an, a, of
+
+word = "bank"
+part_of_speech = "Noun"
+language = "English"
+number_of_clusters = 2
+
+# building model 
+# Dimensionality of the resulting word vectors.
+#more dimensions mean more traiig them, but more generalized
+num_features = 300
+
+# Minimum word count threshold - the smallest set of words we want to recognise 
+min_word_count = 3
```

Comment:
can we define the below variables locally? Or is it must for us to have them globally?
min_word_count, context_size, seed &  downsampling

Reply:
they all are used in one function, so we can push it to just be local variables, will it be easier for us to look for them in a function or just at the top of the code in case of changing them?

Reply:
fine we can have keep them in the top